lib/commands/publish.js

```diff
@@ -204,25 +209,27 @@ class Publish extends BaseCommand {
     }
   }
 
-  async #latestPublishedVersion (spec, registry) {
+  async #registryVersions (spec, registry) {
     try {
       const packument = await pacote.packument(spec, {
         ...this.npm.flatOptions,
         preferOnline: true,
         registry,
       })
       if (typeof packument?.versions === 'undefined') {
-        return null
+        return { versions: [], latest: null }
       }
       const ordered = Object.keys(packument?.versions)
         .flatMap(v => {
           const s = new semver.SemVer(v)
           return s.prerelease.length > 0 ? [] : s
         })
         .sort((a, b) => b.compare(a))
-      return ordered.length >= 1 ? ordered[0].version : null
+      const latest = ordered.length >= 1 ? ordered[0].version : null
+      const versions = ordered.map(v => v.version)
+      return { versions, latest }
```

Comment:
🤔 nit, possibly discussed on https://github.com/npm/cli/pull/7939: `latest` here doesn't refer to the current version with the well-known `latest` tag. It doesn't even refer to the version published last chronologically.

Is there another term we could use? `greatest` or `greatestReleasedVersion` perhaps?